### PR TITLE
work on allow removing canvas

### PR DESCRIPTION
**Pull request by nadav-ab**
breaking change

should be supported in codux,

allows toggling canvas on and off

### CODE REVIEW

File: packages/react-board/src/types.ts [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
+     *
+     * @returns a cleanup function
+     */
+    render: (targetElement: HTMLElement) => Promise<() => void>;
+    version: string;
 }
+
+export interface BoardPlugin {
+    onInit?: (board: IReactBoard) => void;
+    WrapRender?: ({ children }: { children: React.ReactNode; board: IReactBoard }) => React.ReactNode;
```
Comment (PeterShershov) [inline packages/react-board/src/types.ts:56]: maybe just `Wrapper`?

File: packages/react-board/src/setup-stage.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+};
+
+export const defaultCanvasProperties: ICanvasEnvironmentProps = {};
+
+const applyStylesToWindow = (windowStyles: IWindowEnvironmentProps = {}, previousProps: IWindowEnvironmentProps) => {
+    // we revert the changes to previous values when running cleanup
+    previousProps.height = previousProps.height ? window.outerHeight : defaultWindowStyles.height;
+    previousProps.width = previousProps.width ? window.outerWidth : defaultWindowStyles.width;
+
+    window.resizeTo(windowStyles.width || previousProps.width, windowStyles.width || previousProps.width);
```
Comment (PeterShershov) [inline packages/react-board/src/setup-stage.tsx:46]: both parameters are `width`
```suggestion
const width = windowStyles.width || previousProps.width;
const height = windowStyles.height || previousProps.height; 
    window.resizeTo(width, height);
```

File: packages/react-board/src/types.ts [hunk 3]
```diff
@@ -1,42 +1,74 @@
...
-    tags?: string[];
-
+export interface IReactBoard extends IRenderableMetadataBase {
     /** Defines whether the board can be used as a snippet. */
     isSnippet?: boolean;
 
-    /** A function that indicates that board is ready to be snapshotted */
-    readyToSnapshot?: () => Promise<void>;
+    /**
+     * if canvas object is omitted no canvas will be rendered around the element
```
Comment (PeterShershov) [inline packages/react-board/src/types.ts:20]:
```suggestion
     * If the canvas object is omitted, no canvas will be rendered around the element.
```

File: packages/react-board/src/types.ts [hunk 4]
```diff
@@ -1,42 +1,74 @@
...
+     * scripts, etc. These functions run only once before the board is mounted.
+     */
+    setup?: BoardSetupFunction | BoardSetupFunction[] | undefined;
     /** A React component representing the board. */
     Board: React.ComponentType<any>;
+    plugins?: BoardPlugin[];
+    /**
+     * sets the stage for the renderer.
+     * this function has many side effects ( such as effecting window styles and sizes )
+     *
```
Comment (PeterShershov) [inline packages/react-board/src/types.ts:40]:
```suggestion
     * Sets the stage for the renderer. 
     * This function has many side effects, such as affecting window styles and sizes.
     *
```

File: packages/react-board/test/setup-stage.spec.tsx [hunk 5]
```diff
@@ -30,60 +34,80 @@ describe('setupBoardStage', () => {
...
         );
     });
+    it('toggles canvas off and then on', () => {
+        const { updateCanvas, canvas: startCanvas } = setupBoard({
+            canvas: {
+                backgroundColor: 'red',
+            },
+        });
+
+        updateCanvas(undefined);
```
Comment (PeterShershov) [inline packages/react-board/test/setup-stage.spec.tsx:97]: just to make sure it mounts correctly

```suggestion
        expect(startCanvas.isConnected, 'canvas was not created').to.equal(true);
        
        updateCanvas(undefined);
```

**Comment by daomry** [comment]
done by peter

**Pull request closed without merging**